rules/renaming/src/Rector/MethodCall/RenameMethodRector.php

```diff
@@ -166,6 +173,15 @@ private function shouldSkipForExactClassMethodForClassMethod(ClassMethod $classM
             return false;
         }
 
-        return $classMethod->getAttribute(AttributeKey::CLASS_NAME) === $type;
+        $isExactClassMethodForClasssMethod = $classMethod->getAttribute(AttributeKey::CLASS_NAME) === $type;
+        if ($isExactClassMethodForClasssMethod) {
+            return true;
+        }
+
+        if ($newMethodName !== '__invoke') {
```

Comment:
All the magic methods should be handled here, as I comented in the issue:
https://github.com/rectorphp/rector/pull/5511#issuecomment-778503158

Reply:
implemented.